### ISSUE TITLE
Enable auto-merge in this repo

### ISSUE THREAD

**Issue by Alan-Jowett** [microsoft/ebpf-for-windows#1562]
Given that our CI/CD now takes around 30 minutes, it would be helpful to enable [auto-merge](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-auto-merge-for-pull-requests-in-your-repository) in this repo. This would simplify the life of maintainers and permit automatically merging PRs when the CI/CD run completes.

**Comment by Alan-Jowett** [comment]
Agreed not to enable auto-merge.

**Comment by dthaler** [comment]
Along with auto-merge we should look at something like
https://github.com/marketplace/actions/auto-update
so that when one PR is merged, another PR would update and start its CI/CD thus creating a pending queue.

Ideally though we should be able to prioritize PR's so merging one doesn't trigger 15 other ones to be updated in parallel, but rather only trigger the highest priority ones that previously passed CI/CD.
What I often do manually at the moment is label PR's "bug", "tests", "documentation", or "enhancement", and try to merge them in that order (bugs before tests before documentation before enhancements).  Rationale is that merging bugs first can fix random failures affecting all other PRs.  Merging tests can then prevent new bugs being introduced by other PRs.

Would be nice to automate some of that.

**Comment by dthaler** [comment]
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/automatically-merging-a-pull-request is the doc on per-PR auto-merging
